erpnext/accounts/doctype/budget/budget.py

```diff
@@ -14,20 +14,24 @@ class DuplicateBudgetError(frappe.ValidationError): pass
 
 class Budget(Document):
 	def autoname(self):
-		self.name = make_autoname(self.cost_center + "/" + self.fiscal_year + "/.###")
+		self.name = make_autoname(self.get(frappe.scrub(self.budget_against)) + "/" + self.fiscal_year + "/.###")
 
 	def validate(self):
+		if not self.get(frappe.scrub(self.budget_against)):
+			frappe.throw(_("{0} is mandatory").format(self.budget_against))
 		self.validate_duplicate()
```

Comment:
This is because we have to make budget either against project or cost center mandatory, not for both of them.